### PR TITLE
feat(api/conf): dashboard new field own role

### PR DESCRIPTION
**Pull request by abuathier**
## Description

> Creation of relation tables
> ➕  Add a **own_role** field in
> - `GET .../configuration/dashboards`
> - `POST .../configuration/dashboards`
> - `GET .../configuration/dashboards/:id`

Jira: 🏷️ **MON-19409**

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [X] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [X] 23.10.x (master)

<h2> How this pull request can be tested ? </h2>

Look at the Jira ticket.

#### Community contributors & Centreon team

- [X] I have followed the **coding style guidelines** provided by Centreon
- [X] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [X] I have commented my code, especially **hard-to-understand areas** of the PR.
- [X] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/src/Core/Dashboard/Domain/Model/DashboardSharingRoles.php [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+        foreach ($contactGroupShares as $contactGroupShare) {
+            Assertion::isInstanceOf(
+                $contactGroupShare,
+                DashboardContactGroupShare::class,
+                "{$shortName}::contactGroupShares"
+            );
+        }
+
+        $this->theMostPermissiveRole = $this->calculateTheMostPermissiveRole();
```
Comment (abuathier) [inline centreon/src/Core/Dashboard/Domain/Model/DashboardSharingRoles.php:55]: remove computation from construct

File: centreon/src/Core/Dashboard/Domain/Model/DashboardRights.php [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    public function canAccess(): bool
+    {
+        return $this->hasViewerRole();
+    }
+
+    public function hasAdminRole(): bool
+    {
+        return $this->contact->isAdmin()
```
Comment (abuathier) [inline centreon/src/Core/Dashboard/Domain/Model/DashboardRights.php:52]: remove admin

File: centreon/src/Core/Dashboard/Domain/Model/DashboardPanelValidationTrait.php [hunk 3]
```diff
@@ -41,8 +41,8 @@ trait DashboardPanelValidationTrait
     private function ensureValidName(string $name): void
     {
         $shortName = (new \ReflectionClass($this))->getShortName();
-        Assertion::maxLength($name, DashboardPanel::MAX_NAME_LENGTH, $shortName . '::name');
-        Assertion::notEmptyString($name, $shortName . '::name');
+        Assertion::maxLength($name, DashboardPanel::MAX_NAME_LENGTH, $shortName.'::name');
+        Assertion::notEmptyString($name, $shortName.'::name');
```
Comment (abuathier) [inline centreon/src/Core/Dashboard/Domain/Model/DashboardPanelValidationTrait.php:45]: php-cs-fixere

File: centreon/src/Core/Dashboard/Domain/Model/DashboardValidationTrait.php [hunk 4]
```diff
@@ -41,8 +41,8 @@ trait DashboardValidationTrait
     private function ensureValidName(string $name): void
     {
         $shortName = (new \ReflectionClass($this))->getShortName();
-        Assertion::maxLength($name, Dashboard::MAX_NAME_LENGTH, $shortName . '::name');
-        Assertion::notEmptyString($name, $shortName . '::name');
+        Assertion::maxLength($name, Dashboard::MAX_NAME_LENGTH, $shortName.'::name');
+        Assertion::notEmptyString($name, $shortName.'::name');
```
Comment (abuathier) [inline centreon/src/Core/Dashboard/Domain/Model/DashboardValidationTrait.php:45]: php-cs-fixer

File: centreon/src/Core/Dashboard/Infrastructure/API/AddDashboard/AddDashboardController.php [hunk 5]
```diff
@@ -59,7 +59,7 @@ public function __invoke(
              *     description?: ?string
              * } $dataSent
              */
-            $dataSent = $this->validateAndRetrieveDataSent($request, __DIR__ . '/AddDashboardSchema.json');
+            $dataSent = $this->validateAndRetrieveDataSent($request, __DIR__.'/AddDashboardSchema.json');
```
Comment (abuathier) [inline centreon/src/Core/Dashboard/Infrastructure/API/AddDashboard/AddDashboardController.php:62]: php-cs-fixer

File: centreon/src/Core/Dashboard/Infrastructure/API/AddDashboard/AddDashboardSchema.json [hunk 6]
```diff
@@ -1,20 +1,20 @@
...
-            "type": [
-                "string",
-                "null"
-            ]
-        }
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "title": "Add a new dashboard",
+  "type": "object",
+  "required": [
+    "name"
```
Comment (abuathier) [inline centreon/src/Core/Dashboard/Infrastructure/API/AddDashboard/AddDashboardSchema.json:6]: indentation

**Review by jeremyjaouen — APPROVED** [review]

**Review by a-launois — APPROVED** [review]

**Pull request merged**